File: firebase_core_custom/ios/FirebaseCorePlugin.m
#import "FirebaseCorePlugin.h"

@implementation FirebaseCorePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"plugins.flutter.io/firebase_core"
            binaryMessenger:[registrar messenger]];
  FirebaseCorePlugin* instance = [[FirebaseCorePlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"Firebase#initializeApp" isEqualToString:call.method]) {
    [self initializeApp:call result:result];
  } else if ([@"Firebase#apps" isEqualToString:call.method]) {
    [self getApps:result];
  } else {
    result(FlutterMethodNotImplemented);
  }
}

- (void)initializeApp:(FlutterMethodCall*)call result:(FlutterResult)result {
  NSString* appName = call.arguments[@"appName"];
  NSDictionary* options = call.arguments[@"options"];
  
  if (!appName) {
    appName = @"DEFAULT";
  }
  
  NSMutableDictionary* appMap = [NSMutableDictionary dictionary];
  [appMap setValue:appName forKey:@"name"];
  
  if (options) {
    [appMap setValue:options forKey:@"options"];
  } else {
    NSMutableDictionary* defaultOptions = [NSMutableDictionary dictionary];
    [defaultOptions setValue:@"stub-api-key" forKey:@"apiKey"];
    [defaultOptions setValue:@"stub-app-id" forKey:@"appId"];
    [defaultOptions setValue:@"stub-sender-id" forKey:@"messagingSenderId"];
    [defaultOptions setValue:@"stub-project-id" forKey:@"projectId"];
    [appMap setValue:defaultOptions forKey:@"options"];
  }
  
  result(appMap);
}

- (void)getApps:(FlutterResult)result {
  // Return a stub apps list with a default app
  NSMutableArray* apps = [NSMutableArray array];
  
  NSMutableDictionary* defaultApp = [NSMutableDictionary dictionary];
  [defaultApp setValue:@"DEFAULT" forKey:@"name"];
  
  NSMutableDictionary* options = [NSMutableDictionary dictionary];
  [options setValue:@"stub-api-key" forKey:@"apiKey"];
  [options setValue:@"stub-app-id" forKey:@"appId"];
  [options setValue:@"stub-sender-id" forKey:@"messagingSenderId"];
  [options setValue:@"stub-project-id" forKey:@"projectId"];
  
  [defaultApp setValue:options forKey:@"options"];
  [apps addObject:defaultApp];
  
  result(apps);
}

@end 

File: firebase_core_custom/android/src/main/java/io/flutter/plugins/firebase/core/FirebaseCorePlugin.java
package io.flutter.plugins.firebase.core;

import androidx.annotation.NonNull;
import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/** FirebaseCorePlugin */
public class FirebaseCorePlugin implements FlutterPlugin, MethodCallHandler {
  private MethodChannel channel;

  @Override
  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
    channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), "plugins.flutter.io/firebase_core");
    channel.setMethodCallHandler(this);
  }

  @Override
  public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
    if (call.method.equals("Firebase#initializeApp")) {
      initializeApp(call, result);
    } else if (call.method.equals("Firebase#apps")) {
      getApps(result);
    } else {
      result.notImplemented();
    }
  }

  private void initializeApp(MethodCall call, Result result) {
    Map<String, Object> arguments = call.arguments();
    String appName = (String) arguments.get("appName");
    
    if (appName == null) {
      appName = "[DEFAULT]";
    }
    
    // Return a mock initialized app
    Map<String, Object> appMap = new HashMap<>();
    appMap.put("name", appName);
    appMap.put("options", getDefaultOptions());
    
    result.success(appMap);
  }

  private void getApps(Result result) {
    // Return a list with just the default app
    ArrayList<Map<String, Object>> apps = new ArrayList<>();
    
    Map<String, Object> defaultApp = new HashMap<>();
    defaultApp.put("name", "[DEFAULT]");
    defaultApp.put("options", getDefaultOptions());
    
    apps.add(defaultApp);
    
    result.success(apps);
  }
  
  private Map<String, Object> getDefaultOptions() {
    Map<String, Object> options = new HashMap<>();
    options.put("apiKey", "mock-api-key");
    options.put("appId", "mock-app-id");
    options.put("messagingSenderId", "mock-sender-id");
    options.put("projectId", "mock-project-id");
    return options;
  }

  @Override
  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
    channel.setMethodCallHandler(null);
  }
} 